### PR TITLE
Backend refactor arg handling

### PR DESCRIPTION
**Pull request by keenanm500**
This modifies how we accept input into the backend-refactor. Args should be in the same order that they are currently inputted, with the addition of a (currently unused) color target option (NGT, etc). This also displays help info if someone doesn't input data correctly.

### CODE REVIEW

File: backend/src/main.py [hunk 1]
```diff
@@ -18,38 +18,67 @@
...
-    # TODO: stadardize parsing of arguments
+    parser.add_argument('-t', '--target', choices=['NGT', 'APT', 'CCSG', 'CC'], default='NGT', help='The color target reference data')
+
+    """
+    Top left & bottom right of (NGT at the moment) target (pixels)
+    """
+    parser.add_argument('top_left_x', help='Pixel value of the top-left of the color target from the left end of the color-target image')
+    parser.add_argument('bottom_right_x', help='Pixel value of the bottom-right of the color target from the left end of the color-target image')
+    parser.add_argument('top_left_y', help='Pixel value of the top-left of the color target from the top end of the color-target image')
+    parser.add_argument('bottom_right_y', help='Pixel value of the bottom-right of the color target from the top end of the color-target image')
```
Comment (keenanm500) [inline backend/src/main.py:36]: Brendan please verify these descriptions

(I realize now you're gonna be mad at the length of these lines)

**Review by jpakey99 — APPROVED** [review]

File: backend/src/main.py [hunk 2]
```diff
@@ -18,38 +18,67 @@
...
+
+    """
+    Location of white square on target image
+    """
```
Comment (Victoriam7) [inline backend/src/main.py:40]: For consistency it would be better to make these # comments unless they are large multiline comments

Comment (Victoriam7) [reply backend/src/main.py:40]: not a huge deal though

File: backend/src/main.py [hunk 3]
```diff
@@ -18,38 +18,67 @@
...
+    parser.add_argument('white_col', help='Column of selected white patch from left of target')
+    parser.add_argument('white_row', help='Row of selected white patch from top of target')
+
+    images_help = '''
```
Comment (Victoriam7) [inline backend/src/main.py:44]: Move this to constants for the help messages if we stick with those

File: backend/src/main.py [hunk 4]
```diff
@@ -18,38 +18,67 @@
...
+        Additional Images... (A and B)
+    '''
+    parser.add_argument('images', nargs='+', help=images_help)
+    args = parser.parse_args()
+
+    if len(args.images) < 6:
+        parser.print_help()
+        sys.exit(1)
 
     """ Setup packet """
```
Comment (Victoriam7) [inline backend/src/main.py:63]: Packet setup should be extracted to a function so we can base it off of the requested pipeline

**Review by Victoriam7 — CHANGES_REQUESTED** [review]
The usage statement is hard to read. It may be better to override the usage statment since argpars's default is pretty bad. I we go this way, it can be done by adding a -h, --help options with the action

```py
 class HelpAction(argparse._HelpAction):
     """ Print usage statement when -h or --help is passed """
     def __call__(self, parser, namespace, values, option_string=None):
         print(USAGE_STATEMENT)
         parser.exit()
```

**Comment by Victoriam7** [comment]
```py
 15 class Parser(argparse.ArgumentParser):
 16     """ Custom parser to allow for a custom error message """
 17
 18     def error(self, message):
 19         print(USAGE_STATEMENT)
 20         print('\033[91m' + 'error: %s\033[00m' % message)
 21         sys.exit(2)

```

**Review by jpakey99 — DISMISSED** [review]

File: backend/src/parser.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import sys
```
Comment (Victoriam7) [inline backend/src/parser.py:1]: Please add a file level comment to the top of the new file similar to the other files

**Review by Victoriam7 — APPROVED** [review]
This looks really good

**Review by sk4900 — DISMISSED** [review]

**Review by Victoriam7 — APPROVED** [review]
This looks great!

**Review by jpakey99 — APPROVED** [review]

**Review by egc2846 — APPROVED** [review]

**Pull request merged**